src/pkcs11/core_pkcs11_test.c

```diff
@@ -0,0 +1,2647 @@
+        xResult = pxGlobalFunctionList->C_VerifyInit( xGlobalSession, &xMechanism, xPublicKeyHandle );
+        TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to VerifyInit RSA." );
+        xResult = pxGlobalFunctionList->C_Verify( xGlobalSession, xPaddedHash, pkcs11RSA_2048_SIGNATURE_LENGTH, xSignature, xSignatureLength );
+        TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to Verify RSA." );
+    }
+
+    mbedtls_rsa_free( &xRsaContext );
+}
+
+/*-----------------------------------------------------------*/
+
+TEST( Full_PKCS11_RSA, AFQP_DestroyObject )
+{
+    CK_RV xResult = CKR_OK;
+
+    xResult = prvDestroyTestCredentials();
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to destroy RSA credentials." );
+}
+
+/*--------------------------------------------------------*/
+/*-------------- EC Tests ------------------------------- */
+/*--------------------------------------------------------*/
+
+TEST_SETUP( Full_PKCS11_EC )
+{
+    CK_RV xResult;
+
+    xResult = xInitializePKCS11();
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to initialize PKCS #11 module." );
+
+    xResult = xInitializePkcs11Session( &xGlobalSession );
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to open PKCS #11 session." );
+}
+
+/*-----------------------------------------------------------*/
+
+TEST_TEAR_DOWN( Full_PKCS11_EC )
+{
+    CK_RV xResult;
+
+    xResult = pxGlobalFunctionList->C_CloseSession( xGlobalSession );
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to close session." );
+
+    xResult = pxGlobalFunctionList->C_Finalize( NULL );
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to finalize session." );
+}
+
+/*-----------------------------------------------------------*/
+
+TEST_GROUP_RUNNER( Full_PKCS11_EC )
+{
+    #if ( PKCS11_TEST_EC_KEY_SUPPORT == 1 )
+        /* Reset the cryptoki to uninitialize state. */
+        prvBeforeRunningTests();
+
+        #if ( PKCS11_TEST_GENERATE_KEYPAIR_SUPPORT == 1 )
+            RUN_TEST_CASE( Full_PKCS11_EC, AFQP_GenerateKeyPair );
+        #endif
+
+        /* Always run AFQP_CreateObject after AFQP_GenerateKeyPair if the port
+         * supports both. Ports that support object importing have extra checks
+         * as the contents of the private key and public key are well known.
+         */
+        #if ( PKCS11_TEST_IMPORT_PRIVATE_KEY_SUPPORT == 1 )
+            RUN_TEST_CASE( Full_PKCS11_EC, AFQP_CreateObject );
+        #endif
+
+        RUN_TEST_CASE( Full_PKCS11_EC, AFQP_FindObject );
+        RUN_TEST_CASE( Full_PKCS11_EC, AFQP_GetAttributeValue );
+        RUN_TEST_CASE( Full_PKCS11_EC, AFQP_Sign );
+        RUN_TEST_CASE( Full_PKCS11_EC, AFQP_Verify );
+
+        RUN_TEST_CASE( Full_PKCS11_EC, AFQP_FindObjectMultiThread );
+        RUN_TEST_CASE( Full_PKCS11_EC, AFQP_GetAttributeValueMultiThread );
+        RUN_TEST_CASE( Full_PKCS11_EC, AFQP_SignVerifyMultiThread );

```

Comment:
It seems these tests are not run for RSA group ?

Reply:
I will also port the test to RSA test group in new comment.

Reply:
AFQP_GetAttributeValueMultiThread is added to RSA test group. Verify function is still under discussion.